package/src/method-element.js

```diff
@@ -215,14 +216,30 @@ class MethodElement {
         validator.currentElements.push($field)
         // Trigger validation error.
         validator.showErrors(errors)
-        return
+      }
+      else {
+        // The error is not related to a payment field, reconstruct error markup.
+        const settings = Drupal.settings.clientsideValidation.forms[this.form_id].general
+        const $message = $(`<${settings.errorElement} class="${settings.errorClass}">`).text(error.message)
+        const $wrapper = $('#clientsidevalidation-' + this.form_id + '-errors')
+        // Add message to clientside validation wrapper if there is one.
+        if ($wrapper.length) {
+          const $list = $wrapper.find('ul')
+          $message.wrap(`<${settings.wrapper}>`).parent().addClass('stripe-error').appendTo($list)
+          $list.show()
+          $wrapper.show()
+        }
+        // Show message above the payment fieldset in want of a better place.
+        else {
+          $message.addClass('stripe-error').insertBefore(this.$element)
+        }
       }
     }
-    // No clientside validation or error is not related to a payment field.
-    if ($('#messages').length === 0) {
-      $('<div id="messages"><div class="section clearfix"></div></div>').insertAfter('#header')
+    // Without clientside validation render a message above the fieldset.
+    else {
+      const $message = $('<div class="messages error">').text(error.message)
+      $message.addClass('stripe-error').insertBefore(this.$element.closest('form'))
```

Comment:
This is above the form (as stated in the commit message) not the fieldset (as stated in the comment), right?

Reply:
yes ;)